linkcheck/checker/ftpurl.py

```diff
@@ -115,8 +115,9 @@ def cwd(self):
         component.
         """
         path = self.urlparts[2]
-        if isinstance(path, bytes):
-            path = path.decode(self.filename_encoding, 'replace')
+        if self.filename_encoding != "utf-8":
+            path = path.encode(self.filename_encoding, 'replace').decode(
+                self.filename_encoding)
```

Comment:
This change doesn't make sense to me.  Can you drop it from this PR please?

(There might be some logic errors in ftpurl's filename handling, but bugfixes shouldn't hide in PRs that purport to be refactorings.)

Reply:
Dropped that change.

Reply:
Raised issue #450.